resolver/dns_resolver.ml

```diff
@@ -254,6 +254,31 @@ let scrub_it t proto zone edns ts ~signed qtype p =
     Log.warn (fun m -> m "NS didn't like us %a" Rcode.pp e) ;
     `Try_another_ns
 
+let likely_blocked reply =
+  match reply.Packet.data with
+  | `Answer (answ, _auth) ->
+    (* HACK! We assume blocked domains have a certain shape. *)
+    Domain_name.Map.for_all
+      (fun _domain rr ->
+         Rr_map.for_all
+           (function
+             | Rr_map.B (Rr_map.A, (3600l, ips)) ->
+               Ipaddr.V4.Set.equal ips (Ipaddr.V4.(Set.singleton localhost))
+             | Rr_map.B (Rr_map.Aaaa, (3600l, ips)) ->
+               Ipaddr.V6.Set.equal ips (Ipaddr.V6.(Set.singleton localhost))
```

Comment:
The match on 3600 ttl is probably overly specific. I'm also thinking we should check the SOA the same way as in the NXdomain case...